### ISSUE TITLE
FetchOHLCV out of sync Coinbase Pro

### ISSUE THREAD

**Issue by Slurpgoose** [ccxt/ccxt#8400]
- OS: **OSX 10.15.7**
- PLV: **Node v14.15.0**
- CCXT version: **1.41.62**

the OHLCV data for coinbase pro is between 1-5 candles behind realtime. 

expected behavior is that at the end of the minute either

1. the candle will update the close price of an existing candle (in the array) and  then create a new candle for the next minute
2.  or add the missing candle to the dataset when the next candle has started.

the synchronization issue is not consistent. meaning in some cases its 1 candle behind sometimes its 5 candles behind.
the[ pro.coinbase docs ](https://docs.pro.coinbase.com/#get-historic-rates) do not mention this issue.

I have also manually implemented the pro api before discovering this library. so I will test it with the other code to find out if this is a library or exchange issue. if it is an exchange issue I think it still makes sense to at least mention it in the ccxt docs so other users do not make the same mistaken assumption that I have made.

the example I provided calls the ccxt.fetchOHLCV() method on a timer. the data is formatted from an array of arrays to an array of objects. then the max timestamp is retrieved and passed to a function that prints the output to see the variance between date.now() and date.now(timestamp).  
```
const ccxt = require('ccxt');
const exchange = new ccxt.coinbasepro({verbose: false});

const formatData = (data, market=null) => {
    return data.map((
        [time, open, high, low, close, volume]) => {
        return {
            time: time / 1000,
            open,
            high,
            low,
            close,
            volume,
            market: market,
        };
    });
};

const printTimeStamp = (last) => {
    const current = new Date(last * 1000);
    const now = new Date();
    console.log(`
       coinbase: ${current.getHours()}:${current.getMinutes()}:${current.getSeconds()}
        now: ${now.getHours()}:${now.getMinutes()}:${now.getSeconds()}

        cb: ${current)}
        now: ${now}
    `);
};

const getData = async(market='BTC/USD') => {
    const data = await exchange.fetchOHLCV(market, '1m');
    let cleaned = formatData(data, market);
    const timestamps = cleaned.map(i => i.time);
    return timestamps;
};

const runTest = async() => {
    const timestamps = await getData();
    const last = Math.max(...timestamps);
    printTimeStamp(last);
};

setInterval(() => runTest(), 2000);
```

script Logging Output
```
  coinbase: 1:57:0 //hh:mm:ss
  now: 1:58:3  //hh:mm:ss

  cb: Fri Feb 05 2021 01:57:00 GMT-0500 (Eastern Standard Time) 
  now: Fri Feb 05 2021 01:58:03 GMT-0500 (Eastern Standard Time)
```

```
  coinbase: 1:52:0 //hh:mm:ss
   now: 1:56:25 //hh:mm:ss

   cb: Fri Feb 05 2021 01:52:00 GMT-0500 (Eastern Standard Time)
   now: Fri Feb 05 2021 01:56:25 GMT-0500 (Eastern Standard Time)
```

**Comment by kroitor** [comment]
Alternatively, you can collect the first-order data yourself and build your tickers and OHLCVs with less lag in the userland, see the `build_ohlcvc` helper method that makes candles from trades.

**Comment by Slurpgoose** [comment]
@kroitor thanks so much for the response this makes complete sense. 

I wish it clearly phrased in coinbase's docs that their endpoint has such a significant lag. I have been using their web-socket feed but I needed some method to have access to candles incase my server was down or if sockets disconnected and I was using the historical endpoint as a source of truth so a couple times an hour I would resync local candles using that endpoint.

so basically there was multiple scenarios where my system was making trades on stale data. I will definitely take a look at the `build_ohlcv` function as well. thanks!

**Comment by kroitor** [comment]
@Slurpgoose yeah, basically, using the first-order trades is as close as you can get to realtime OHLCVs with the REST APIs. If you want it even "more real" time – you might want to consider WS APIs: https://ccxt.pro/
